src/app/features/courses/components/search/search.component.ts

```diff
@@ -1,17 +1,39 @@
-import { Component, EventEmitter, Output } from '@angular/core';
+import { Component, EventEmitter, Output, ViewChild, ElementRef, OnInit, OnDestroy } from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { debounceTime, filter, takeUntil } from 'rxjs/operators';
+
 
 @Component({
   selector: 'app-search',
   templateUrl: './search.component.html',
   styleUrls: ['./search.component.scss']
 })
-export class SearchComponent {
+export class SearchComponent implements OnInit, OnDestroy {
+
+  @Output() makeSearchQuery = new EventEmitter<string>();
+  @ViewChild('searchInput') searchInputRef: ElementRef;
+
+  searchInputValue = '';
+  private destroy$ = new Subject();
+
+  ngOnInit() {
+    const searchInputObservable = fromEvent(this.searchInputRef.nativeElement, 'input');
 
-  @Output() makeSearchQuery: EventEmitter<string> = new EventEmitter();
-  searchQuery = '';
+    searchInputObservable
+      .pipe(
+        debounceTime(1000),
+        filter((e: any) => e.target.value.length >= 3),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(e => this.search());
+  }
+
+  search() {
+    this.makeSearchQuery.emit(this.searchInputValue);
+  }
 
-  search(): void {
-    this.makeSearchQuery.emit(this.searchQuery);
-    this.searchQuery = '';
+  ngOnDestroy() {
+    this.destroy$.next();
+    this.destroy$.complete();
```

Comment:
look at this, it's better way to handle subscription https://github.com/ngneat/until-destroy